subcommands/keys/tuf_utils.go

```diff
@@ -260,3 +267,61 @@ func RemoveUnusedKeys(root *client.AtsTufRoot) {
 		}
 	}
 }
+
+func checkTufRootUpdatesStatus(updates client.TufRootUpdates, forUpdate bool) (
+	curCiRoot, newCiRoot *client.AtsTufRoot,
+) {
+	switch updates.Status {
+	case client.TufRootUpdatesStatusNone:
+		subcommands.DieNotNil(errors.New(`There are no TUF root updates in progress.
+Please, run "fioctl keys tuf updates init" to start over.`))
+	case client.TufRootUpdatesStatusStarted:
+		break
+	case client.TufRootUpdatesStatusApplying:
+		if forUpdate {
+			subcommands.DieNotNil(errors.New(
+				"No modifications to TUF root updates allowed why they are being applied.",
+			))
+		}
+	default:
+		subcommands.DieNotNil(fmt.Errorf("Unexpected TUF root updates status: %s", updates.Status))
+	}
+
+	if updates.Current != nil && updates.Current.CiRoot != nil {
+		subcommands.DieNotNil(json.Unmarshal(updates.Current.CiRoot, &curCiRoot))
+	}
+	if curCiRoot == nil {
+		subcommands.DieNotNil(errors.New("Current TUF CI root not set. Please, report a bug."))
+	}
+	if updates.Updated != nil && updates.Updated.CiRoot != nil {
+		subcommands.DieNotNil(json.Unmarshal(updates.Updated.CiRoot, &newCiRoot))
+	}
+	if newCiRoot == nil {
+		subcommands.DieNotNil(errors.New("Updated TUF CI root not set. Please, report a bug."))
+	}
+	return
+}
+
+func genProdTufRoot(ciRoot *client.AtsTufRoot) (prodRoot *client.AtsTufRoot) {
+	// Deep copy in Golang is hard; use the marshal-unmarshal trick
+	body, err := json.Marshal(ciRoot)
+	subcommands.DieNotNil(err)
+	subcommands.DieNotNil(json.Unmarshal(body, &prodRoot))
+	prodRoot.Signed.Roles["targets"].Threshold = 2
```

Comment:
Do we support only two targets role keys for the prod atm?

Reply:
Yes, at the moment there are always two TUF targets keys for prod: 1 online and 1 offline.

The support for multiple offline keys is out of scope for this feature (planned for the future).

Reply:
cool.